tools/src/prebuilds/ExpoModulesCorePrebuilder.ts

```diff
@@ -0,0 +1,175 @@
+import assert from 'assert';
+import fs from 'fs-extra';
+import path from 'path';
+
+import { podInstallAsync } from '../CocoaPods';
+import { getExpoRepositoryRootDir } from '../Directories';
+import logger from '../Logger';
+import { Package } from '../Packages';
+import * as XcodeGen from './XcodeGen';
+import { ProjectSpec } from './XcodeGen.types';
+import XcodeProject, {
+  flavorToFrameworkPath,
+  spreadArgs,
+  SHARED_DERIVED_DATA_DIR,
+} from './XcodeProject';
+import { Flavor, Framework, XcodebuildSettings } from './XcodeProject.types';
+
+// Generates working files out from expo-modules-core folder and prevents CocoaPods generates unnecessary files inside expo-modules-core folder
+const OUT_OF_TREE_WORKING_DIR = path.join(getExpoRepositoryRootDir(), 'prebuild-ExpoModulesCore');
```

Comment:
Does it need to happen inside the repository folder? Can we prepare it in the temp dir?